book/src/proposals/handle-duplicate-block.md

```diff
@@ -0,0 +1,119 @@
+# Leader Duplicate Block Slashing
+
+This design describes how the cluster slashes leaders that produce duplicate
+blocks.
+
+Leaders that produce multiple blocks for the same slot increase the number of
+potential forks that the cluster has to resolve.
+
+## Primitives
+1. gossip_root: Nodes now gossip their current root
+2. gossip_duplicate_slots: Nodes can gossip up to `N` duplicate slot proofs.
+3. `DUPLICATE_THRESHOLD`: The minimum percentage of stake that needs to vote on a fork with version `X` of a duplicate slot, in order for that fork to become votable.
+
+## Protocol
+1. When WindowStage detects a duplicate slot proof `P`, it checks the new `gossip_root` to see if `<= 33%` of the nodes have rooted a slot `S >= P`. If so, it pushes a proof to `gossip_duplicate_slots` to gossip. WindowStage then signals ReplayStage about this duplicate slot `S`. These proofs can be purged from gossip once the validator sees > 2/3 of people gossiping roots `R > S`.
+
+2. When ReplayStage receives the signal for a duplicate slot `S` from `1)` above, the validator monitors gossip and replay waiting for`>= DUPLICATE_THRESHOLD` votes for the same hash which implies the same version of the slot. If this conditon is met for some version with hash `H` of slot `S`, this is then known as the `duplicate_confirmed` version of the slot.
+
+Before a duplciate slot `S` is `duplicate_confirmed`, it's first excluded rom the vote candidate set in the fork choice rules. In addition, ReplayStage also resets PoH to the *latest* ancestor of the *earliest* `non-duplicate/confirmed_duplicate_slot`, so that block generation can start happening on the earliest known *safe* block. 
+
+Some notes about the `DUPLICATE_THRESHOLD`. In the cases below, assume `DUPLICATE_THRESHOLD = 52`:
+
+a) If less than `2 * DUPLICATE_THRESHOLD - 1` percentage of the network is malicious, then there can only be one such `duplicate_confirmed` version of the slot. With `DUPLICATE_THRESHOLD = 52`, this is
+a malcious tolerance of `4%`
+
+b) The liveness of the network is at most `1 - DUPLICATE_THRESHOLD - SWITCH_THRESHOLD`. This is because if you need at least `SWITCH_THRESHOLD` percentage of the stake voting on a different fork in order to switch off of a duplicate fork that has `< DUPLICATE_THRESHOLD` stake voting on it, and is *not* `duplicate_confirmed`. For `DUPLICATE_THRESHOLD = 52` and `DUPLICATE_THRESHOLD = 38`, this implies a liveness tolerance of `10%`. 
```

Comment:
The tradeoffs here are important, would welcome input on how to improve these if possible. One thought was lowering the switching threshold from 38% to 33% which would allow us to bump `DUPLICATE_THRESHOLD` to 57%, and still have `10%` liveness, and also `14%` malicious/double vote tolerance.